Implementing MarketStackAPI Client + Show Use Case

alexlee27 opened the pull request:
I was having some git issues with the previous branch, so I deleted the old one and created this one instead.

MarketStackClient.java uses the end-of-day API endpoint of MarketStack to access the historical price data of a stock (https://marketstack.com/documentation#historical_data).

I have also modified pom.xml to add dependencies that MarketStackClient.java uses.

I set up the files for the show use case, which are still empty, which is why I have started this pull request as a draft.

alexlee27 commented:
![image](https://github.com/wkukka1/PortfolioTracker/assets/105831093/61cc0eb0-c8d2-4b84-b0a4-d7edf8ebdd2b)
The API client and the show plot use case are done! The portfolio's net worth over the past 1000 days is plotted and returned to the logged-in screen. I also had to reformat the frontend layout as it was acting a little funny.

* For weekends and holidays, there is no stock information, so technically the portfolio's net worth is zero on those days. If this wasn't taken into consideration, the graph would drop to 0 every weekend/holiday, which would have made the plot ugly. I got around this by simply taking the net worth of the previous day (or the most recent day with stock information), so the graph doesn't drop to 0.

K-Logo reviewed (approved):
Great work! There were a few minor things that I would like to see changed before approving (such as removing the redundant file, if it's truly not needed). 
I liked your use of Javadoc, it was very helpful.
Some other things to consider include:

- Including a comment at the top of a file to explain what the file is doing. In paraticular, the StockPriceClientImpl.java file's role wasn't obvious from the start.
- I would also like to see some test cases added, that pass and show the line coverage.
Good work!

alexlee27 commented:
Note: Since we have changed the API endpoint from the one from MarketStack to the one from Alpha Vantage, I have updated the name of this PR accordingly.
